torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -399,6 +399,126 @@ class FunctionInliner : public IRMutator {
   std::unordered_set<const Var*> func_var_set_;
 };
 
+/**
+ * Actual algorithm for "inlining" of rand().  Bind variables to rand() calls.
+ */
+class RandomInliner : public IRMutator {
```

Comment:
I'm wondering whether RandomInliner should inherit from FunctionInliner, and extend the additional functionalities.

Reply:
Yeah I think it should.  I started off by copying the impl on the assumption it would be more different, but it's really not.